### PR TITLE
Update init command to start ws server, implement of connect command to join a ws game server, broadcasting messages during actions and poker table log poc

### PR DESCRIPTION
**Pull request by Abhay-2811**
1. On init command, the aze-cli spawns a broadcast server where game related messages would be published.
2. On connect command with --url arg, players can connect to broadcast server and listen to game events.
3. These commands store was URL in a temp file called (ws_url.json)
4. Broadcast messages when an action is taken by player.
5. Implemented `stats` command and poker table log poc.
6. Fixed Broadcast bugs and implemented `\stats` endpoint so players can get stats from game account.

### CODE REVIEW

**Comment by Abhay-2811** [comment]
Hey @nlok5923 , pushed the new changes implementing the broadcast message on actions.

File: cli/src/actions/mod.rs [hunk 1]
```diff
@@ -17,14 +17,23 @@ use miden_objects::{
...
 ) -> Result<GameActionResponse, String> {
     let mut client: AzeClient = create_aze_client();
     let player_account_id = AccountId::try_from(player_id).unwrap();
     let game_account_id = AccountId::try_from(game_id).unwrap();
+    let ws_url = Ws_config::load(ws_config_path).url.unwrap();
+    
+    //TODO: need a better way for this
+    let parts: Vec<&str> = ws_url.split('/').collect();
+    let game_id_hex = parts.last().map(|s| s.to_string()).unwrap();
+    let _ = broadcast_message(game_id_hex.clone(), ws_url.clone(), format!("Player: {} plays raise by amount: {}", player_id, amount.unwrap())).await;
```
Comment (nlok5923) [inline cli/src/actions/mod.rs]: Anyways we need to improve the messages, Will do this a bit later maybe

File: cli/src/main.rs [hunk 2]
```diff
@@ -20,15 +22,17 @@ enum Commands {
...
     Register(RegisterCmd),
+    Connect(ConnectCmd),
+    Stats(StatsCmd)
 }
 
 #[tokio::main]
 async fn main() {
     let args = Args::parse();
-
+    let ws_config_path = std::path::PathBuf::from("ws_config.json");
```
Comment (nlok5923) [inline cli/src/main.rs:34]: Are we assuming user to have a `ws_config.json` with `url` field empty in it for the time being. Untill the game is not initialized.

File: cli/src/stats.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+             {}\n\
+             {}\n\
+             {}\n\
+             {}\n\
+             {}\n\
+             {}\n\
+             {}\n\
+             {}\n\
+             {}",
+            Blue.bold().paint("+-----------------------------+"),
```
Comment (nlok5923) [inline cli/src/stats.rs:44]: We need to make the values dynamic here. Also this won't be affected by the screen resolutions right ?

File: lib/src/broadcast/mod.rs [hunk 4]
```diff
@@ -0,0 +1,179 @@
+use futures_util::{SinkExt, StreamExt};
+use get_if_addrs::get_if_addrs;
+use log::{error, info};
```
Comment (nlok5923) [inline lib/src/broadcast/mod.rs:3]: Can you remove this extra `.DS_store` mac files or maybe add this in `.gitignore`

File: lib/src/client/mod.rs [hunk 5]
```diff
@@ -365,9 +365,9 @@ impl<N: NodeRpcClient, R: FeltRng, S: Store, A: TransactionAuthenticator> AzeGam
         account_storage_mode: AccountStorageMode,
         slot_data: GameStorageSlotData,
     ) -> Result<(Account, Word), ClientError> {
-        if let AccountStorageMode::OnChain = account_storage_mode {
-            todo!("Recording the account on chain is not supported yet");
-        }
+        // if let AccountStorageMode::OnChain = account_storage_mode {
```
Comment (nlok5923) [inline lib/src/client/mod.rs:575]: @Created-for-a-purpose did we added this condition sometime back ? If yes then i didn't remember why

File: lib/src/utils/mod.rs [hunk 6]
```diff
@@ -179,3 +183,66 @@ pub async fn setup_accounts(
...
+        println!("Message successfully published");
+        Ok(())
+    } else {
+        let status = response.status();
+        let error_text = response.text().await?;
+        eprintln!("Failed to publish message: {} - {}", status, error_text);
+        Err(format!("Failed to publish message: {} - {}", status, error_text).into())
+    }
+}
+
```
Comment (nlok5923) [inline lib/src/utils/mod.rs:258]: Remove these empty spaces.

**Review by nlok5923 — CHANGES_REQUESTED** [review]

File: lib/src/client/mod.rs (hunk 5 above)
Comment (Created-for-a-purpose) [reply lib/src/client/mod.rs:575]: No, I don't remember adding this.

File: cli/src/actions/mod.rs (hunk 1 above)
Comment (Abhay-2811) [reply cli/src/actions/mod.rs]: Refactored this

File: cli/src/main.rs (hunk 2 above)
Comment (Abhay-2811) [reply cli/src/main.rs:34]: No if ws_config.json isn't present it'll create one

File: cli/src/stats.rs (hunk 3 above)
Comment (Abhay-2811) [reply cli/src/stats.rs:44]: Done in new commit

File: lib/src/broadcast/mod.rs (hunk 4 above)
Comment (Abhay-2811) [reply lib/src/broadcast/mod.rs:3]: Done

**Pull request merged**